### DISCUSSION TITLE
Custom SSG Cache Key

### DISCUSSION THREAD

**Discussion by j** [vercel/next.js#23723] [Help]
### Describe the feature you'd like to request

Currently, it appears that the SSG cache key is dependent on the path, AMP, and locale.  It'd be nice to be able to add to this cache key if possible.

Why?

We'd like to be able to append cache key based off of a request header or query parameter.  I.E. `X-Variant-ID` header, and if undefined, just cache as normal.

### Describe the solution you'd like

Create new function for pages to configure within the page:

```typescript
export const getStaticPropsCacheKey: GetStaticPropsCacheKey = async ({ req, disable }) => {
    // don't cache when "authenticated"
    if (req.headers['Authorization']) {
        return disable();
    }

    return req.headers['X-SSG-Cache-Key'];
}
```

Alternatively could add something to the global config.

### Describe alternatives you've considered

Currently have to use a custom server, but lose the benefits of everything else.

Also considered a root folder for the path, though this requires a proxy in front to re-write / remove the path.

**Comment by BowlingX** [comment]
This would be awesome. I'm currently rendering pages slightly different based on the user agent (mobile, tablet, desktop), so there might be different data fetched based on the client in some cases.
